lib/extensions/require_nested.rb

```diff
@@ -6,23 +6,19 @@ def require_nested(name)
     filename = "#{self}::#{name}".underscore
     filename = name.to_s.underscore if self == Object
     if Rails.application.config.cache_classes
+      raise LoadError, "No such file to load -- #{filename}" unless ActiveSupport::Dependencies.search_for_file(filename)
```

Comment:
This is the main part of the PR.

Reply:
do we want to do this in development and test only?
to not incur the hit to the disk?

Or do we consider the fstat calls across the load path minor?

Reply:
I think that's the current issue @kbrock, right now we're setting `cache_classes = true` in test and prod which is causing this issue to not be found in test, but simultaneously I would think we want test to be as close to prod as possible in this regard.

FWIW the direct call to `require_dependency` in the `else` branch will fail in the case addressed by this PR.

Reply:
@carbonin aah, so this check will never be run in development. thanks (not what I wanted, but more clear)